### ISSUE TITLE
net.http.download_file doesn't work with non text files.

### ISSUE THREAD

**Issue by Cyklan** [vlang/v#5490]
**V version:** 0.1.28
**OS:** 
- Ubuntu 18.04.3 LTS (WSL)
- Windows 10

**What did you do?**
I wrote some code to download a `.png` image from imgur.
```v
import net.http
import os

fn main() {
	args := os.args
	http.download_file(args[1], "./test.png")
	println("done!")
}
```

**What did you expect to see?**
I expected the image to appear on my hard drive, labeled `test.png`. Opening the file should show the image in its entirety.
 
**What did you see instead?**
I received a file called `test.png`, but it's only 1kb in size, and its binary content is basically empty
```
�PNG

```

imgur link for reference: https://i.imgur.com/p6HNwxl.png

I've also prepared a demo gif.
![broken_download](https://user-images.githubusercontent.com/37187056/85628033-c02cad80-b66f-11ea-812f-261b40061164.gif)

**Comment by sudo-nice** [comment]
The `http.get()` function fails to work with binary data.
```v
import net.http

fn main() {
  img := 'https://raw.githubusercontent.com/vlang/v/master/examples/tetris/screenshot.png'
  // but this will work
  // img := 'https://raw.githubusercontent.com/donnisnoni95/v-logo/master/dist/v-logo.svg'
  resp := http.get(img) or {
    panic(err)
  }
  println(resp.status_code) // 200
  println(resp.text)        // not what you expected
}
```

**Comment by JalonSolov** [comment]
Looking at the code, yes, http.download_file() will only work with text files.

There are more problems than that, though.  I expanded the example a bit, and found out it _does_ get the whole file, but even using write_bytes isn't writing out the whole file.
```
import net.http
import os

fn main() {
	img := 'https://raw.githubusercontent.com/vlang/v/master/examples/tetris/screenshot.png'
	out := 'screenshot.png'

	println('getting url = $img')
	s := http.get(img) or {
		panic(err)
	}
	println('file size = $s.text.len')
	mut file := os.create(out) or {
		panic(err)
	}
	file.write_bytes(s.text, s.text.len)
	file.close()
	println('wrote ${os.file_size(out)} bytes to $out')
}
```

And here's a sample run:
```
jalon@Ripper:~/git/v$ v run foo.v
getting url = https://raw.githubusercontent.com/vlang/v/master/examples/tetris/screenshot.png
file size = 98186
wrote 8192 bytes to screenshot.png
jalon@Ripper:~/git/v$
```

**Comment by sudo-nice** [comment]
That `8192 bytes` looks suspicious.

**Comment by JalonSolov** [comment]
It's worse... I wound up running it multiple times, without changing the code, and it seemed to randomly either write 4096 or 8192.  :-\

**Comment by div72** [comment]
Fixed with #5646.
